### DISCUSSION TITLE
Hdr

### DISCUSSION THREAD

**Discussion by oMarvixo** [VirtualDrivers/Virtual-Display-Driver#4] [Ideas]
can we make a option for hdr ? 

i love the idea for the virtual display 

only hdr is missing for the perfect virtual display solution 

i added a 2960 x 1848 pixels for my galaxy tab s9 ultra it works like a charm and looks beautiful 

the only thing i miss right now is hdr

**Comment by Nitroedge** [comment]
I would so love HDR as I am sure many would. I tried ordering different dummy plugs (both DP and HDMI) with no luck getting 4k 60Hz HDR to work.
My use case is: PC in office (non-HDR monitor) running Sunshine ---> nVidia Shield in family room (Moonlight + Samsung OLED HDR TV) 
Trying to stream HDR content to my HDR TV using nVidia Shield
My PC in office cannot recognize or offer Sunshine 4k HDR.

**Reply by oMarvixo** [reply]
did you tried this one it says 4k 60hz hdr ?

https://www.amazon.com/FUERAN-Plug-Virtual-Emulator-3840x2160-60-3840x2160/dp/B0C174243H

**Reply by Nitroedge** [reply]
No I didn't, I just ordered it and will be able to test it out this weekend and report back. Thanks for the link! :)

**Reply by kayti** [reply]
If it is only about 4k 60Hz HDR You can use any dummy plug and with this 
https://www.monitortests.com/forum/Thread-EDID-DisplayID-Writer
Overwrite its default EDID.
Works perfectly. Problem with HDR is that, it is ignored in custom EDID made with tol like CRU. BUT flashing this custom EDID directly to dummy plug makes it work. I use it in my current setup for HDR.

And as i said, it works for 4k 60 Hz HDR. But sadly, it doesn't for high refresh rate and high resolution like 4k 120 Hz etc. This tool can only have 1 additional block of EDID data. 4k 60Hz, HDR, 7.1 sounds etc are all configurable within one specific block of data, so it works well. But high refresh/resolution needs another type of datablock and writing both (HDR + High refresh rate) is not possible because this 1 datablock limitation. In this case it is HDR OR Highr refresh rate.

**Reply by Nitroedge** [reply]
> did you tried this one it says 4k 60hz hdr ?
> 
> https://www.amazon.com/FUERAN-Plug-Virtual-Emulator-3840x2160-60-3840x2160/dp/B0C174243H

I received the HDR dummy plug you recommended... and immediately my Windows 10 detected it, I was able to go Display Settings and enable HDR for the first time on the 2nd display adapter...

Should I force Sunshine to that display //DISPLAY06 (example) and then I'm confused as to how Moonlight on my Nvidia Shield will know to play the game on the 2nd dummy HDR monitor from my RTX 3080.... I haven't tried this yet... just wondering how to configure the Display Settings even in Windows as do I need to change the 2nd dummy monitor to "Make this my main display"?

Duplicating the displays just resulted in a lower resolution with no HDR option since my main 3440x1440 monitor doesn't support HDR....

**Reply by oMarvixo** [reply]
the problem is you plug the dongle in and turn off the normal display and it should work directly just plug and play that how it works by my 

you could try to fore sunshine to the display how that works i dont know exactly 
just try everthing if you tried everything and it doesnt work we can look up to gother to troubleshoot whats going on and how we can get it working 

have a nice day and dont hassle to ask questions im here to help

**Reply by Nitroedge** [reply]
Thanks for your help.

I got it working 100% now.

When I launch Moonlight on the Shield, the Sunshine on my PC launches and:
1) Makes Display 1 (Ultrawide) disabled
2) Makes Display 2 (Dummy HDR HDMI) enabled and primary display
3) Launches Playnite fullscreen

Then when I exit Moonlight, the reverse occurs and all works awesome.

**Reply by homehtha** [reply]
Can you write steps needed to make it work?

**Comment by superraddm** [comment]
Hi my use case. Remote windows 11 PC with a 3090TI. Streaming to a Windows 10 PC connected to an LG 55CX5LB via HDMI 2.2. Managed to install the virtual display driver, and get 4k at 120Hz/FPS/.However when I enable HDR on the virtual monitor the colours are all washed out and too bright. Do I need somehow to add the HDR profile specific to my TV? Using Sunlight as streaming server on the remote machine and Moonlight client on the client PC. Thanks.
![image](https://github.com/itsmikethetech/Virtual-Display-Driver/assets/17827279/c5f1b6fa-e731-4ce0-b8fb-a7315f2b44ff)

**Reply by Nitroedge** [reply]
This has happened to me in the past but my use case is different, on the nVidia Shield the client Moonlight (connecting to my PC running Sunshine) will say "PC GPU recognized as a SDR display" or something similar and I've had to go into the Moonlight settings and check the HDR support box right at the bottom of the options screen.

Its usually when the client displays the host as a standard monitor when the actual monitor is in HDR mode from the Windows Settings.

**Comment by superraddm** [comment]
Thanks I have a lot more trial and error to do. Worth noting that in Windows 10 (client machine) HDR is enabled for the LG TV in Windows settings. So all in all i have HEVC Main and Main10, AV1 Main 8-bit and Main 10 bit profiles advertised in Sunshine on the host. HDR enabled on the virtual display on client machine Windows 11 settings. HDR enabled in Moonlight on the client. And HDR enabled in windows 10 display settings on the client.
![image](https://github.com/itsmikethetech/Virtual-Display-Driver/assets/17827279/033cf806-6538-4fdb-9797-31fd2154a96a)

![image](https://github.com/itsmikethetech/Virtual-Display-Driver/assets/17827279/f6ac659e-2847-4685-93c7-1c244c42870b)

Could something be amiss here?

**Comment by Nitroedge** [comment]
That looks right, and in the Moonlight client on the Win 10 machine you have HDR ticked as well right?
And you don't have any other multiple monitors connected?

One thing you could try (this is what I do on a multiple display setup where my main monitor is non-HDR and I have a HDMI dummy plug inserted into my PC which is how I get the HDR on my nVidia Shield).

I always before launching Moonlight on my client, I press Windows key + P and then choose "second screen only" so that my Moonlight client doesn't get confused receiving an initial SDR monitor 1 signal then flicking to the dummy plug HDR signal.

I had major problems when my Sunshine PC had mutli-monitors active.... that was when Moonlight would show my 2nd HDR dummy plug screen with washed out colors and brightness cranked.  When I disable the Sunshine PC primary monitor and isolated it to only output on the 2nd screen (HDR dummy plug), Moonlight would then recognize only that screen and show HDR properly colors and the Sunshine PC desktop.

When I had problems, during Moonlight startup on my nVidia Shield, below the connecting information it would say that infamous "PC SDR only detected" message.

**Comment by superraddm** [comment]
Thanks, yep client machine has HDR ticked.

I'll try disconnecting the second monitor on the client machine, but I have a *interesting* (complicated) setup. For reasons I can't remember. I have one HDMI 2.1 cable going from client machine to LGTV for 4K 120hz. I then have a second HDMI 1.4 cable into my Denon AV AMP for audio. I think I did it this way because the Denon doesn't have e-arc, and can't do 4K 120hz. So I'm thinking if I disconnect that second monitor (Denon). I'll lose audio. But it's defo worth testing. I think it's more and more likely I'll need to get the dummy plug. Do you have one you can recommend? 

Its interesting you get a HDR status message on your Moonlight app on the Shield. I don't think there's one present on the Windows version, though it would be very helpful.

**Comment by superraddm** [comment]
Still working my way through this. But FWIW. 2 screenshots from a remote desktop app captured on a non HDR screen showing Client HDR  physical monitor (LGTV known to work and be correct) and Host HDR Virtual monitor. I understand these aren't accurate representations, but the fact that they present very different colour information makes me think something is amiss at the virtual monitor side.
1. Client
![image](https://github.com/itsmikethetech/Virtual-Display-Driver/assets/17827279/8887a347-21e3-435c-b147-99735f03473c)
2. Remote Host
![image](https://github.com/itsmikethetech/Virtual-Display-Driver/assets/17827279/ddac2a9d-c24f-4eed-9369-db8c651d6e7c)

**Comment by Nitroedge** [comment]
Ya thats what I saw as well.
I should mention and this is super important, my fault I didn't catch this sooner!
I never got HDR to work with the virtual driver. I was on this thread talking about how the virttual driver did work for me but I said I did find a HDR dummy plug that worked 100% for me.

So I don't know if others got HDR working with the virtual driver, I never could and didn't google for that since I found a hardware solution which worked.

The dummy plug I bought was this one and can confirm it works awesome with HDR:
https://www.amazon.com/FUERAN-Plug-Virtual-Emulator-3840x2160-60-3840x2160/dp/B0C174243H

I'm sorry I didn't mention that earlier as I must have misread a reply. But hopefully this will help with your solution! Thanks!

**Reply by superraddm** [reply]
Hi @Nitroedge Sorry for my delay in replying. I've been knee deep and testing and problem solving. As mentioned, unfortunately the Virtual Display Driver didn't work for me. But I do now have it working. 120hz/FPS HDR 4k. I tried the solution discussed on this reddit thread: https://www.reddit.com/r/MoonlightStreaming/comments/143kgf1/4k_120hz_hdr_edid_emulator/ Buying the EDID and Display port 1.4 to HDMI plug. But, my client machine only had a 1080 card, with no HDMI 2.1 output. I managed to buy a 3070ti from eBay for £300, and since installing in the client can now stream from the host PC, over the internet, to my client, plugged into the LG and Denon AMP with 5.1 surround, HDR, 120hz/FPS and in 4k. Thank you for all your help.

**Comment by itsmikethetech** [comment]
We now have HDR in the Windows 11 version, but not Windows 10. Closing for now, but happy to reopen if other issues arise.

**Reply by ming019** [reply]
Hi @itsmikethetech, much appreciate your hardwork to make HDR possible for VDD. Is there a reason why only WIndows 11 get HDR for VDD but not 10?

**Comment by joeyoropesa-dev** [comment]
I would like to know is it possible to use physical built-in monitor to display data from second monitor. Basically switching display roles so that virtual display will behave like real display. Maybe some kind of EDID hack that will put data received from virtual monitor to real monitor.

My device is Dell Vostro 3400 so I hope that this can help with finding correct EDID hack to achieve this goal. Making my device HDR compatible : )

**Reply by kayti** [reply]
It will not work that way, HDR is more physical trait than pure software, there is needed proper HDR capable hardware, other way the colors would be washed out.

**Reply by joeyoropesa-dev** [reply]
> It will not work that way, HDR is more physical trait than pure software, there is needed proper HDR capable hardware, other way the colors would be washed out.

So if this driver could somehow connect to physical display to transmit signal there, display would start to behave too strange and unusable.


But could at least some kind of driver to behave like dummy display to real display (basically becoming software-driver based projector) that will mirror data from virtual display just like OBS is doing in it's software method. So, HDR options would be clickable and how OBS optimize HDR pictures, similar thing could be made for projector mode in dummy display driver for physical displays.

Or this will not do the trick too...

**Comment by itsmikethetech** [comment]
![image](https://github.com/user-attachments/assets/73e084fd-03d3-4020-a9c6-2cf7d1d88c58)

I do have a display preview on the Patreon that allows you to view virtual displays while you use them. What specific functionality were you looking for?

https://www.patreon.com/posts/display-preview-124318492?utm_medium=clipboard_copy&utm_source=copyLink&utm_campaign=postshare_creator&utm_content=join_link

That being said, just like OBS, it's a software method. All other methods will also be software due to monitor hardware limitations.
